src/main/kotlin/io/hsar/wh40k/combatsimulator/model/MoveAction.kt

```diff
@@ -0,0 +1,24 @@
+package io.hsar.wh40k.combatsimulator.model
+
+interface  MoveAction {
+    fun getMovementRange(agilityBonus: Short): Short
+    fun isValidMovementPath(startPoint: MapPosition, endPoint: MapPosition): Boolean
+}
+
+object HalfMoveAction: MoveAction {
```

Comment:
`HalfMoveAction` and others should inherit from `TurnAction` ideally, as well as implementing `MoveAction` like you have it. This allows us to look up the cost of each move action when considering whether to take it.

See `io/hsar/wh40k/combatsimulator/logic/TurnAction`